pkgs/unified_analytics/test/events_with_mock_test.dart

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:clock/clock.dart';
+import 'package:file/file.dart';
+import 'package:file/memory.dart';
+import 'package:path/path.dart' as p;
+import 'package:test/test.dart';
+
+import 'package:unified_analytics/src/constants.dart';
+import 'package:unified_analytics/src/enums.dart';
+import 'package:unified_analytics/src/survey_handler.dart';
+import 'package:unified_analytics/unified_analytics.dart';
+
+import 'src/mock_analytics.dart';
+
+void main() {
+  // The mocked analytics instance can be used to ensure events
+  // are being sent when invoking methods on the `Analytics` instance
+
+  late MockAnalytics mockAnalytics;
+  late FileSystem fs;
+  late Directory homeDirectory;
+  late File clientIdFile;
+
+  /// Survey to load into the mock instance to fetch
+  ///
+  /// The 1.0 sample rate means that this will always show
+  /// up from the method to fetch available surveys
+  final testSurvey = Survey(
+    'uniqueId',
+    'url',
+    DateTime(2022, 1, 1),
+    DateTime(2022, 12, 31),
+    'description',
+    10,
+    'moreInfoUrl',
+    1.0, // 100% sample rate
+    <Condition>[],
+  );
+
+  /// Test event that will need to be sent since surveys won't
+  /// be fetched until at least one event is logged in the persisted
+  /// log file on disk
+  final testEvent = Event.hotReloadTime(timeMs: 10);
+
+  setUp(() async {
+    fs = MemoryFileSystem.test(style: FileSystemStyle.posix);
+    homeDirectory = fs.directory('home');
+
+    // Write the client ID file out so that we don't get
+    // a randomly assigned id for this test generated within
+    // the analytics constructor
+    clientIdFile = fs.file(p.join(
+      homeDirectory.path,
+      kDartToolDirectoryName,
+      kClientIdFileName,
+    ));
+    clientIdFile.createSync(recursive: true);
+    clientIdFile.writeAsStringSync('string1');
+
+    final initialAnalytics = Analytics.test(
+      tool: DashTool.flutterTool,
+      homeDirectory: homeDirectory,
+      measurementId: 'measurementId',
+      apiSecret: 'apiSecret',
+      dartVersion: 'dartVersion',
+      toolsMessageVersion: 1,
+      fs: fs,
+      platform: DevicePlatform.macos,
+    );
+    initialAnalytics.clientShowedMessage();
+
+    // Recreate a second instance since events cannot be sent on
+    // the first run
+    await withClock(Clock.fixed(DateTime(2022, 3, 3)), () async {
```

Comment:
Does this closure need to be async? There's nothing being awaited.

Reply:
Ah certainly does not need to `async` great catch! Updating that now along with the other nites